### PR TITLE
Release 2023.10.0

### PR DESCRIPTION
**Pull request by charlesbluca**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by charlesbluca** [comment]
maturin:
    - 1.1

**Comment by charlesbluca** [comment]
@conda-forge-admin, please rerender

**Comment by charlesbluca** [comment]
@conda-forge-admin, please rerender

**Comment by jakirkham** [comment]
Looks like `x86_64` flags are being passed to ARM jobs (on [CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=798445&view=logs&j=e9afaa34-1a0f-534e-78fc-fae528ccd915&t=869ab379-d63c-50ae-655f-69018fb07995&l=567 )), which understandably causes an error

```
error: unrecognized command-line option '-m64'
```

File: recipe/build.sh
Comment (jakirkham) [inline recipe/build.sh:58]: Looks like this file doesn't exit on macOS x86_64. From [CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=798559&view=logs&j=9c5ef928-2cd6-52e5-dbe6-9d173a7d951b&t=20c71c51-4b27-578b-485d-06ade2de1d00&l=192 ):

```
+ sed -i.bak s,aarch64,arm64,g /Users/runner/miniforge3/conda-bld/dask-sql_1696630065483/_build_env/venv/lib/os-patch.py
sed: /Users/runner/miniforge3/conda-bld/dask-sql_1696630065483/_build_env/venv/lib/os-patch.py: No such file or directory
```

<br>

Should we add a conditional around this?

File: recipe/build.sh
Comment (jakirkham) [inline recipe/build.sh:59]: Guessing this is also absent on macOS x86_64

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

**Comment by ayushdg** [comment]
Is it also possible to reduce the number of threads/jobs while building arm with `-j 1` or `-j 2` ? If this is indeed memory related this might help.
```
maturin build -h
Options:
  -j, --jobs <N>                            Number of parallel jobs, defaults to # of CPUs
```

File: recipe/build.sh
Comment (jakirkham) [inline recipe/build.sh]: To limit the number of parallel jobs used in line with [limits set on conda-forge CI]( https://github.com/conda-forge/conda-forge-ci-setup-feedstock/blob/4dd3bebe7951df2bb8ed54064194c6c815dee16d/recipe/run_conda_forge_build_setup_linux#L27 )

```suggestion
maturin build -vv -j "${CPU_COUNT}" --release --strip --manylinux off --interpreter="${PYTHON}" "${_xtra_maturin_args[@]}"
```

Comment (jakirkham) [reply recipe/build.sh]: Thanks @ayushdg ! 🙏

Can you please check my syntax above (please feel free to edit as well)?

Comment (ayushdg) [reply recipe/build.sh]: This generally makes sense to me. Once the current builds are complete it might be worth trying with a smaller value than `CPU_COUNT` if the run fails.

Comment (jakirkham) [reply recipe/build.sh]: Thanks Ayush! 🙏

Yeah `CPU_COUNT` would be smaller than the default. We have it set to `2`. Usually that is good enough, but we can try `1` if not

**Comment by jakirkham** [comment]
Looks like Linux ARM packages are now passing with the latest changes! 🎉

Waiting for Charles to take a look over things tomorrow and do any last cleanup

Probably can merge soon after

**Comment by jakirkham** [comment]
It's worth noting the memory usage of these builds is quite high (on [CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=801186&view=logs&j=696704cc-6fef-57a3-ea36-f27779b8cd5e&t=43aed047-1209-5a61-2f15-186544abf594&l=2412 )). So we may want to keep an eye on that (and see if there are ways we can reduce that amount)

```
Maximum memory usage observed: 5.7G
```

File: recipe/build.sh
Comment (ayushdg) [reply recipe/build.sh]: Ah I see, thanks for confirming I missed that.

**Comment by charlesbluca** [comment]
Thanks for the help here @jakirkham and @ayushdg! Sounds like two approaches we could take to reducing memory pressure here further in case we find builds continuing to fail are:

- relying on cross compilation for `linux-aarch64` using `--zig`, which is what we do for the [wheel builds](https://github.com/dask-contrib/dask-sql/actions/runs/6433023555/job/17469180596)
- lowering the number of parallel jobs during the `maturin` command itself 

Considering the time pressures, I'm okay with merging this in mostly as is (removed the commented code and some environment variables that I don't think should have any impact on the builds) and following up on this work in the repo and a follow up build

**Comment by charlesbluca** [comment]
@conda-forge-admin, please rerender

**Comment by jakirkham** [comment]
Do we need to drop Windows? It was passing before above

**Comment by charlesbluca** [comment]
There seem to be some intermittent [failures](https://github.com/conda-forge/dask-sql-feedstock/runs/17609031450) that seem to imply that things might fail once merged - am I correct in assuming that or does conda-forge use beefier machines for the actual package building/publishing so that this wouldn't be an issue?

**Comment by jakirkham** [comment]
Let's try the same thing with Windows that we did with Linux

https://conda-forge.org/docs/maintainer/conda_forge_yml.html#azure

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render

**Comment by jakirkham** [comment]
Looks like this is now passing! 🎉

**Review by ayushdg — APPROVED** [review]

**Review by jakirkham — APPROVED** [review]

**Pull request merged**
